lib/inferno/dsl/fhir_evaluation/rules/all_search_parameters_have_examples.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require_relative '../reference_extractor'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        class AllSearchParametersHaveExamples < Rule
+          attr_accessor :unused_resource_urls
+
+          def check(context)
+            unless ENV.fetch('FHIRPATH_URL')
+              message = 'FHIRPATH_URL is not found. Skipping rule AllSearchParametersHaveExamples.'
+              result = EvaluationResult.new(message, severity: 'warning', rule: self)
+              context.add_result result
+              return
+            end
+
+            @unused_resource_urls = []
+            search_params = context.ig.to_hash[:resources_by_type]['SearchParameter']
+
+            get_unused_resource_urls(context.ig.to_hash[:resources_by_type]['SearchParameter']) do |param|
+              param_is_used?(param, context)
+            end
+
+            if unused_resource_urls.any?
+              message = "Found SearchParameters with no searchable data: \n\t#{unused_resource_urls.join(' ,')}"
+              result = EvaluationResult.new(message, rule: self)
+            elsif !search_params.empty?
+              message = 'All SearchParameters have examples'
+              result = EvaluationResult.new(message, severity: 'success', rule: self)
+            else
+              message = 'IG contains no SearchParameter'
+              result = EvaluationResult.new(message, severity: 'information', rule: self)
+            end
+
+            context.add_result result
+          end
+
+          def get_unused_resource_urls(ig_data, &resource_filter)
+            ig_data.each do |resource|
+              unused_resource_urls.push resource.url unless resource_filter.call(resource)
+            end
+          end
+
+          def param_is_used?(param, context)
+            # Assume that all params have an expression (fhirpath)
+            # This is not guaranteed since the field is 0..1
+            # but without it there's no other way to select a value
+            return true unless param.expression
```

Comment:
Here and in line 66, what is the reasoning behind reporting that search parameters are used if they can't be evaluated?

Reply:
I originally wrote this part and to be honest I don't remember exactly. I think the intent in the error condition on line 66 was to avoid double-reporting the search param in the output messages, but now that you point it out I agree this one on line 51 is just silently letting things through. I'm ok with changing both of these to false

Reply:
I don't think these should pass.